spec/ruby/core/file/dirname_spec.rb

```diff
@@ -24,6 +24,19 @@
 
     it "raises ArgumentError if the level is negative" do
       -> {File.dirname('/home/jason', -1)}.should raise_error(ArgumentError)
+
+    it "returns the exact same object when passed 0 as the level" do
+      obj = "path"
+      File.dirname(obj,0).should .equal?(obj)
+
+      obj = mock("to_path")
+      def obj.to_path
+        "path"
+      end
+      File.dirname(obj,0).should .equal?(obj)
+
+      obj = Object.new
+      File.dirname(obj,0).should .equal?(obj)
```

Comment:
Good catch!

I noticed there is another missing test case - when level is greater than actual file name directories number, I mean
```ruby
File.dirname('/a/b', 10) # => '/'
```